plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/glue/TrinoGlueCatalog.java

```diff
@@ -0,0 +1,520 @@
+            if (database.getLocationUri() != null) {
+                metadata.put(LOCATION_PROPERTY, database.getLocationUri());
+            }
+            if (database.getParameters() != null) {
+                metadata.putAll(database.getParameters());
+            }
+            return metadata.buildOrThrow();
+        }
+        catch (EntityNotFoundException e) {
+            throw new SchemaNotFoundException(namespace);
+        }
+        catch (AmazonServiceException e) {
+            throw new TrinoException(ICEBERG_CATALOG_ERROR, e);
+        }
+    }
+
+    @Override
+    public Optional<TrinoPrincipal> getNamespacePrincipal(ConnectorSession session, String namespace)
+    {
+        return Optional.empty();
+    }
+
+    @Override
+    public void createNamespace(ConnectorSession session, String namespace, Map<String, Object> properties, TrinoPrincipal owner)
+    {
+        try {
+            CreateDatabaseRequest createDatabaseRequest = new CreateDatabaseRequest()
+                    .withDatabaseInput(createDatabaseInput(namespace, properties));
+            catalogId.ifPresent(createDatabaseRequest::setCatalogId);
+            stats.getCreateDatabase().call(() -> glueClient.createDatabase(createDatabaseRequest));
+        }
+        catch (AlreadyExistsException e) {
+            throw new SchemaAlreadyExistsException(namespace);
+        }
+        catch (AmazonServiceException e) {
+            throw new TrinoException(ICEBERG_CATALOG_ERROR, e);
+        }
+    }
+
+    private DatabaseInput createDatabaseInput(String namespace, Map<String, Object> properties)
+    {
+        DatabaseInput databaseInput = new DatabaseInput().withName(namespace);
+        Object location = properties.get(LOCATION_PROPERTY);
+        if (location != null) {
+            databaseInput.setLocationUri((String) location);
+        }
+
+        return databaseInput;
+    }
+
+    @Override
+    public void setNamespacePrincipal(ConnectorSession session, String namespace, TrinoPrincipal principal)
+    {
+        throw new TrinoException(NOT_SUPPORTED, "setNamespacePrincipal is not supported for Iceberg Glue catalogs");
+    }
+
+    @Override
+    public void renameNamespace(ConnectorSession session, String source, String target)
+    {
+        throw new TrinoException(NOT_SUPPORTED, "renameNamespace is not supported for Iceberg Glue catalogs");
+    }
+
+    @Override
+    public List<SchemaTableName> listTables(ConnectorSession session, Optional<String> namespace)
+    {
+        try {
+            return stats.getGetAllTables().call(() -> {
+                List<String> namespaces = namespace.map(List::of).orElseGet(() -> listNamespaces(session));
+                return namespaces.stream()
+                        .flatMap(glueNamespace -> {
+                            try {
+                                GetTablesRequest getTablesRequest = new GetTablesRequest().withDatabaseName(glueNamespace);
```

Comment:
can you set `GetTablesRequest` without `withDatabaseName` to get all tables from all namespaces?

Reply:
No, databaseName is required unfortunately